catching polymorphic type ‘class std::runtime_error’ by value

Was building `bionic` branch on Arch GNU/Linux, faced this issue:
```
[ 42%] Building CXX object src/modules/Unity/Application/CMakeFiles/unityapplicationplugin.dir/upstart/taskcontroller.cpp.o
/mnt/antihype/Projects/AUR/_Unity8/_Unity8-Arch/qtmir-git/src/qtmir/src/modules/Unity/Application/upstart/taskcontroller.cpp: In function ‘std::shared_ptr<ubuntu::app_launch::Application> qtmir::upstart::{anonymous}::createApp(const QString&, std::shared_ptr<ubuntu::app_launch::Registry>)’:
/mnt/antihype/Projects/AUR/_Unity8/_Unity8-Arch/qtmir-git/src/qtmir/src/modules/Unity/Application/upstart/taskcontroller.cpp:81:33: error: catching polymorphic type ‘class std::runtime_error’ by value [-Werror=catch-value=]
     } catch (std::runtime_error e) {
                                 ^
cc1plus: all warnings being treated as errors
make[2]: *** [src/modules/Unity/Application/CMakeFiles/unityapplicationplugin.dir/build.make:278: src/modules/Unity/Application/CMakeFiles/unityapplicationplugin.dir/upstart/taskcontroller.cpp.o] Error 1
```

--- a/src/modules/Unity/Application/upstart/taskcontroller.cpp
+++ b/src/modules/Unity/Application/upstart/taskcontroller.cpp
@@ -78,7 +78,7 @@ std::shared_ptr<ual::Application> createApp(const QString &inputAppId, std::shar
 
     try {
         return ual::Application::create(appId, registry);
-    } catch (std::runtime_error e) {
+    } catch (std::runtime_error& e) {
         qCDebug(QTMIR_APPLICATIONS) << "ApplicationController::createApp - UAL failed to create app for appId"
                                     << inputAppId << ":" << e.what();
         return {};

